### DISCUSSION TITLE
Should we change the package name?

### DISCUSSION THREAD

**Discussion by anranruye** [ant-design-blazor/ant-design-blazor#2166] [General]
In vs nuget package manager, when search with the key word "blazor", the "AntDesign" package ranks very low in the result, although the package does have a "blazor" tag. I think this is very bad for promotion and exposure.

The casue may be that the package name doesn't contains "blazor".

Should we have a new package name?

@ant-design-blazor/committer Let me know what you think.

**Comment by LeaFrock** [comment]
Agreed. The current package name `AntDesign` informs nothing related to `Blazor`, and might confuse new developers. `Blazor.AntDesign`, which is similar to another common blazor package `Blazor.LocalStorage`, or `AntDesign.Blazor`， is better at least.

**Reply by anranruye** [reply]
I like `AntDesign.Blazor` or `AntDesignBlazor` better. In fact `AntDesignBlazor` and `AntBlazor` are used before but then abandoned.

**Reply by LeaFrock** [reply]
LGTM 😆

**Comment by mutouzdl** [comment]
I rarely look for third-party packages on Nuget. Usually, I know the third-party packages through other means and then go to Nuget to find them.
So the name of the package doesn't really matter to me.

**Reply by anranruye** [reply]
确实有很多人是这样工作的. 但是会使用nuget包管理器探索一些新鲜东西的人应该也不在少数?

我在搜索引擎中以搜索"blazor component lib", 在前排的文章中看到了对antblazor的推荐, antblazor要比我此前想象中更流行和受到欢迎.

这样看来nuget包管理器中的搜索结果排名确实不那么重要, 此前 @ElderJames 表示过暂时不用修改包名, 我想这也是他的底气所在.

问题是, 这个排名实在是太低了(100+), 低到找不到的程度, 无论如何这不能算是好事. 可以推测这是因为nuget的相关性判断中包名的权重要高于tag标签和下载量, 否则应该不至于此.

-----------------------------------------

Many people do work like this But there are not a few people who can use nuget package manager to explore something new?
I searched "blazor component lib" in the search engine, and saw the recommendation of ant blazor in the article in the front row. Ant blazor is more popular and popular than I thought before
In this way, the ranking of search results in nuget package manager is really not so important. Earlier @ elder James said he didn't need to modify the package name for the time being. I think this is also his strength
The problem is that this ranking is too low (100 +), too low to be found. In any case, this is not a good thing It can be inferred that this is because the weight of package name in nuget correlation judgment is higher than that of tag tag and download volume, otherwise it should not be so

**Reply by LeaFrock** [reply]
Another reason is that, why does a developer choose this package?  More for `Ant Design UI` ? There have been already multiple implements(Vue/React/Angular) of it. So I think more for `Blazor` tech stack. 

The current package name `AntDesign` really confuses me at the first time. I do think `AntDesign.Blazor` is better for users, which informs all that this package(project) combines `AntDesign` and `Blazor` .

**Reply by mutouzdl** [reply]
两位的意思我能理解，其实我也不反对改名称，AntDesign.Blazor也挺好。
不过，我想表达的是，包名字真的没太大影响，真正重要的是这个项目能不能流传出去。也许未来大家都是通过百度、知乎、CSDN等各种平台得知AntDesignBlazor这个项目，这些才是最重要的，就像Ant Design React、Vue等，我们都是通过官方文档一步步跟着去安装使用的，至于这个包名是什么，真的无所谓。
最关键的是，这个项目要足够强大、足够好用、不会突然停止维护，否则，哪怕它排在Nuget的第一位，也是没什么帮助的。
ElderJames的想法估计也和我差不多，没必要太关注包名。

总而言之的话，在改名字这件事情上，我并不反对，你们提出的名字也挺好的，只不过，名字可能没那么重要而已。

---

I understand what you mean, and I'm not opposed to changing the name. Antdesign.blazor is fine too.
However, what I want to say is that the name of the package really doesn't matter. What really matters is whether the project gets out. In the future, most of us will learn about AntDesignBlazor through Baidu, Zhihu, CSDN and other platforms. These will be the most important. Like AntDesign React, Vue, etc., we will install and use it step by step through official documents.
Crucially, the project has to be strong enough and usable enough not to suddenly stop maintenance, or it won't help even if it's at the top of Nuget's list.
ElderJames probably thinks the same as me, so don't pay too much attention to the package name.
All in all, I have no objection to changing the name, and the name you proposed is fine, but it may not be that important.

**Comment by ElderJames** [comment]
My initial thought was to just search for `AntDesign` in NuGet and it would bring up our project, which I think is better than `AntBlazor`.
I agree with @mutouzdl  that what matters more is quality, visibility and the amount of use. 

If we need to increase the search ranking of NuGet, we can send a separate package called `AntDesign.Blazor` with the same content as `AntDesign`.

**Comment by anranruye** [comment]
I think we need more feedback about this.
